data-runtime/src/main/java/io/micronaut/data/runtime/intercept/AbstractQueryInterceptor.java

```diff
@@ -758,6 +754,33 @@ protected boolean isNumber(@Nullable Class<?> type) {
         return Number.class.isAssignableFrom(type);
     }
 
+    /**
+     * Sets the property value for given persistent property of the {@link BeanWrapper} if property
+     * present in given parameter values and property not readonly or generated.
+     *
+     * @param wrapper the bean wrapper
+     * @param prop the persistent property
+     * @param parameterValues the parameter value map
+     */
+    private static void setProperty(BeanWrapper<Object> wrapper, PersistentProperty prop, Map<String, Object> parameterValues) {
+        if (!prop.isReadOnly() && !prop.isGenerated()) {
```

Comment:
So if readonly or generated (most of id fields) then it will be skipped. Only non generated ids will be checked